hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/metadata/AbstractDatanodeStore.java

```diff
@@ -89,11 +91,14 @@ protected AbstractDatanodeStore(ConfigurationSource config, long containerID,
       AbstractDatanodeDBDefinition dbDef, boolean openReadOnly)
       throws IOException {
 
+    dbProfile = config.getEnum(HDDS_DB_PROFILE,
+        HDDS_DEFAULT_DB_PROFILE);
+
     // The same config instance is used on each datanode, so we can share the
     // corresponding column family options, providing a single shared cache
     // for all containers on a datanode.
     if (!OPTIONS_CACHE.containsKey(config)) {
-      OPTIONS_CACHE.put(config, buildColumnFamilyOptions(config));
+      OPTIONS_CACHE.put(config, buildColumnFamilyOptions(config, dbProfile));
```

Comment:
Nit: instead of passing `dbProfile` member variable, can you please make `buildColumnFamilyOptions` non-static?

Reply:
oh, thanks for your advice, I'll fix it and try to add a simple test case~